subprojects/core/src/main/java/org/gradle/internal/vfs/RoutingVirtualFileSystem.java

```diff
@@ -95,6 +95,11 @@ public void updateWithKnownSnapshot(CompleteFileSystemLocationSnapshot snapshot)
         getVirtualFileSystemFor(snapshot.getAbsolutePath()).updateWithKnownSnapshot(snapshot);
     }
 
+    @Override
+    public VirtualFileSystemStatistics getStatistics() {
+        return gradleUserHomeVirtualFileSystem.getStatistics();
```

Comment:
Why return the user home stats here? Because we only need the stats when VFS retention is on?

Reply:
Yeah. I could select the user home one or the build session one depending on whether retention is on or off. WDYT?

Reply:
Is there a point in generating retention stats when retention is off? Seems like this method shouldn't be here maybe but on some other type?